rr/ai/ai_manager.py

```diff
@@ -135,14 +140,18 @@ def __init__(
             self,
             model,
             disp_width,
-            disp_height,
-            on_new_prediction_cb,
-            on_new_postprocess_cb):
+            disp_height):
 
         super().__init__(model, disp_width, disp_height)
 
-        self.on_new_prediction_cb = on_new_prediction_cb
-        self.on_new_postprocess_cb = on_new_postprocess_cb
+        self.on_new_prediction_cb_ = None
+
+        self._on_new_prediction_mutex = Lock()
+
+    def install_callback(self, on_new_prediction_cb_):
+        self._on_new_prediction_mutex.acquire()
```

Comment:
We don't need a mutex here. There's not really a shared resource, this can be happening simultaneously.